src/solver/conv_hip_implicit_gemm_wrw_v4r4_xdlops.cpp

```diff
@@ -0,0 +1,1074 @@
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/solver.hpp>
+#include <miopen/conv/invokers/impl_gemm.hpp>
+#include <miopen/conv/wrw_invoke_params.hpp>
+#include <miopen/handle.hpp>
+#include <miopen/generic_search.hpp>
+#include <miopen/hip_build_utils.hpp>
+#include "implicitgemm_util.hpp"
+
+#include <miopen/stringutils.hpp>
+#include <miopen/tensor_ops.hpp>
+#include <miopen/implicitgemm_params.hpp>
+
+namespace miopen {
+namespace solver {
+
+PerformanceImplicitGemmWrwV4R4Xdlops::PerformanceImplicitGemmWrwV4R4Xdlops()
+    : PerformanceImplicitGemmWrwV4R4Xdlops::PerformanceImplicitGemmWrwV4R4Xdlops(
+          64, 64, 2, 32, 32, 4, false, false, 4)
+{
+}
+
+PerformanceImplicitGemmWrwV4R4Xdlops::PerformanceImplicitGemmWrwV4R4Xdlops(
+    int GemmMPerBlock_,
+    int GemmNPerBlock_,
+    int GemmKPerBlock_,
+    int GemmMPerWave_,
+    int GemmNPerWave_,
+    int GemmKPack_,
+    bool GemmAThreadCopyMoreGemmK_,
+    bool GemmBThreadCopyMoreGemmKPack_,
+    int GemmKBlocks_)
+    : GemmMPerBlock(GemmMPerBlock_),
+      GemmNPerBlock(GemmNPerBlock_),
+      GemmKPerBlock(GemmKPerBlock_),
+      GemmMPerWave(GemmMPerWave_),
+      GemmNPerWave(GemmNPerWave_),
+      GemmKPack(GemmKPack_),
+      GemmAThreadCopyMoreGemmK(GemmAThreadCopyMoreGemmK_),
+      GemmBThreadCopyMoreGemmKPack(GemmBThreadCopyMoreGemmKPack_),
+      GemmKBlocks(GemmKBlocks_)
+{
+}
+
+bool PerformanceImplicitGemmWrwV4R4Xdlops::
+operator==(const PerformanceImplicitGemmWrwV4R4Xdlops& other) const
+{
+    // clang-format off
+    return GemmMPerBlock == other.GemmMPerBlock
+        && GemmNPerBlock == other.GemmNPerBlock
+        && GemmKPerBlock == other.GemmKPerBlock
+        && GemmMPerWave == other.GemmMPerWave
+        && GemmNPerWave == other.GemmNPerWave
+        && GemmKPack == other.GemmKPack 
+        && GemmAThreadCopyMoreGemmK  == other.GemmAThreadCopyMoreGemmK
+        && GemmBThreadCopyMoreGemmKPack  == other.GemmBThreadCopyMoreGemmKPack
+        && GemmKBlocks == other.GemmKBlocks;
+    // clang-format on
+}
+
+bool PerformanceImplicitGemmWrwV4R4Xdlops::SetNextValue()
+{
+    do
+    {
+        // list performance parameters in reverse order, in order for tuning to iterate over the
+        // range in normal order
+        if(!NextFlag<false, true>(GemmBThreadCopyMoreGemmKPack))
+            break;
+        if(!NextFlag<false, false>(GemmAThreadCopyMoreGemmK))
+            break;
+        if(!NextTwoPower<4, 8>(GemmKPack))
+            break;
+        if(!NextTwoPower<32, 128>(GemmNPerWave))
+            break;
+        if(!NextTwoPower<32, 128>(GemmMPerWave))
+            break;
+        if(!NextTwoPower<2, 8>(GemmKPerBlock))
+            break;
+        if(!NextTwoPower<64, 256>(GemmNPerBlock))
+            break;
+        if(!NextTwoPower<64, 256>(GemmMPerBlock))
+            break;
+        if(!NextTwoPower<1, 128>(GemmKBlocks))
```

Comment:
Do we need such large `GemmKBlocks`? Do we have a heuristic to avoid try `GemmKBlocks` from 1? Otherwise, there are too many tuning combinations, making tuning ridiculous slow.

Reply:
If removing `GemmKBlocks ` does not harm performance, it will be great.